includes/class-wc-payments-webhook-reliability-service.php

```diff
@@ -0,0 +1,91 @@
+<?php
+// phpcs:ignoreFile TODO using this only when working.
+// TODO - consider singleton?
+
+/**
+ * WC_Payments_Webhook_Reliability_Service class
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+use WCPay\Logger;
+
+/**
+ * Improve webhook reliability by fetching failed events from the server.
+ * And process them with ActionScheduler
+ */
+class WC_Payments_Webhook_Reliability_Service {
+	const FETCHING_EVENTS_LIMIT = 10;
+	const WEBHOOK_FETCH_EVENTS_ACTION = 'wcpay_webhook_fetch_events';
+	const WEBHOOK_PROCESS_EVENT_ACTION = 'wcpay_webhook_process_event';
+
+	/**
+	 * Client for making requests to the WooCommerce Payments API.
+	 *
+	 * @var WC_Payments_API_Client
+	 */
+	private $payments_api_client;
+
+	/**
+	 * WC_Payments_Action_Scheduler_Service.
+	 *
+	 * @var WC_Payments_Action_Scheduler_Service
+	 */
+	private $action_scheduler_service;
+
+	/**
+	 * WC_Payments_Webhook_Reliability_Service constructor.
+	 *
+	 * @param WC_Payments_API_Client $payments_api_client Payments API client.
+	 */
+	public function __construct( WC_Payments_API_Client $payments_api_client ) {
+		$this->payments_api_client = $payments_api_client;
+		$this->action_scheduler_service = new WC_Payments_Action_Scheduler_Service( $this->payments_api_client );
+
+		add_action( self::WEBHOOK_FETCH_EVENTS_ACTION, [$this, 'fetch_events'] );
+		add_action( self::WEBHOOK_PROCESS_EVENT_ACTION, [$this, 'process_event'] );
+		add_action( 'woocommerce_payments_account_refreshed', [ $this, 'maybe_schedule_fetch_events'] );
+	}
+
+	private function fetch_events() {
+		$payload = $this->payments_api_client->get_failed_webhook_events();
+
+		$remaining_failed_events = (int) $payload['failed_webhook_count'] ?? 0; // TODO - maybe `failed_webhook_count` (int) or `has_more` (bool). See server PR 1633
```

Comment:
We will not be using `COUNT(*)` in DB queries, see `$count + 1` in 1633-gh-Automattic/woocommerce-payments-server/files#diff-91becda04a3d12e72f51da7e3f60e14114fc78c8c1090c93af5de85c19b77155R91. `has_more` will simply be boolean in order to know whether to schedule another batch or not.